vaadin-spring/pom.xml

```diff
@@ -155,6 +155,14 @@
             <version>${vaadin.flow.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.security</groupId>
+            <artifactId>spring-security-oauth2-jose</artifactId>
```

Comment:
Shouldn't both dependency be optional? Otherwise they gonna leak to all applications

Reply:
Hi @knoobie, thanks for the comment. Do you see any other problems other than having 2 unnecessary dependencies?
The situation is that we want to enable stateless authentication for Fusion by default. The Spring add-on is shared by both Fusion and Flow, so that if we include these as optional dependencies, then the Fusion app would need to add those dependencies manually, which might be more inconvenient than having 2 unused dependencies for Flow apps. What do you think?

Reply:
Unnecessary and probably conflicting dependencies if flow apps already have some other oauth2-dependencies. Normally flow apps have e.g. `oauth2-client` as dependency but not `oauth2-resource-server`. 

I would suggest to create a separate modules; `spring-flow` and `spring-fusion`. Otherwise changes needed for one framework, leak to the other framework and cause problems for complex customer projects. I'm already exluding `fusion-endpoint` from my flow only apps to reduce transitive dependencies.

Reply:
Example how this could be done with minimal user interruption: 

Split vaadin-spring into two modules; one for each framework. Creating a vaadin-*-spring-boot-starter for each of them and a update the vaadin-spring-boot-starter to have both dependencies. Once this is released; the vaadin-spring-boot-starter could be deprecated and both framework starters could be added independently or combined and be promoted both on e.g. start.spring.io (Vaadin Flow and Vaadin Fusion) (Ping @mstahv because he has done the communication in the past with spring if I remember correctly)

Reply:
@knoobie has a point here. I'd though just make a separate starter for Fusion and keep the old one for Flow (and clean it if there are already obsolete deps for plain Flow users). There isn't that many Fusion apps yet, so it is easy to communicate to them that they need to change the starter dependency. Those using both can add both deps.

Reply:
Thanks for the suggestion, marked those as optional dependencies.